### PR TITLE
[harvester doc] doc about how to expand pvc size

### PR DESCRIPTION
**Pull request by w13915984028**
doc about how to expand pvc size

Signed-off-by: Jian Wang <w13915984028@gmail.com>

View the final update in v1.1:
https://deploy-preview-212--harvester-preview.netlify.app/next/troubleshooting/monitoring#expand-pvvolume-size

View the final update in v1.0:
https://deploy-preview-212--harvester-preview.netlify.app/v1.0/troubleshooting/monitoring#expand-pvvolume-size

**Please review the change in v1.1.  All changes will be back-ported into v1.0.**

issue:
https://github.com/harvester/harvester/issues/2724

### CODE REVIEW

File: docs/troubleshooting/monitoring.md [hunk 1]
```diff
@@ -69,3 +69,107 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+# kubectl get volume -A
+NAMESPACE         NAME                                       STATE      ROBUSTNESS   SCHEDULED   SIZE          NODE     AGE
+longhorn-system   pvc-1b2fbbe9-14b1-4a65-941a-7d5645a89977   attached   degraded                 5368709120    harv31   43h
+longhorn-system   pvc-7c6dcb61-51a9-4a38-b4c5-acaa11788978   attached   degraded                 53687091200   harv31   43h
+longhorn-system   pvc-b2b2c07c-f7cd-4965-90e6-ac3319597bf7   attached   degraded                 2147483648    harv31   43h
+```
+
+### Scale down deployment
+
+The `Volume` needs to be detached at first.
```
Comment (vickyhella) [inline docs/troubleshooting/monitoring.md:118]:
```suggestion
To scale down a deployment, you need to detach the `Volume` first.
```

File: docs/troubleshooting/monitoring.md [hunk 2]
```diff
@@ -69,3 +69,107 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+
+Following example is against the PVC claimed by `rancher-monitoring-grafana`.
```
Comment (vickyhella) [inline docs/troubleshooting/monitoring.md:120]:
```suggestion
The example below is against the PVC claimed by `rancher-monitoring-grafana`.
```

File: docs/troubleshooting/monitoring.md [hunk 3]
```diff
@@ -69,3 +69,107 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+
+Find out the `deployment` in namespace `cattle-monitoring-system`.
```
Comment (vickyhella) [inline docs/troubleshooting/monitoring.md:122]:
```suggestion
Find out the `deployment` in the namespace `cattle-monitoring-system`.
```

File: docs/troubleshooting/monitoring.md [hunk 4]
```diff
@@ -69,3 +69,107 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
+rancher-monitoring-grafana              1/1     1            1           43h
+rancher-monitoring-kube-state-metrics   1/1     1            1           43h
+rancher-monitoring-operator             1/1     1            1           43h
+rancher-monitoring-prometheus-adapter   1/1     1            1           43h
+```
+
+Scale down the deployment `rancher-monitoring-grafana` to 0.
+
+`kubectl scale --replicas=0 deployment/rancher-monitoring-grafana  -n cattle-monitoring-system`
```
Comment (vickyhella) [inline docs/troubleshooting/monitoring.md:135]: Please use ``` to wrap this line of code

File: docs/troubleshooting/monitoring.md [hunk 5]
```diff
@@ -69,3 +69,107 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+
+Check deployment and volume.
```
Comment (vickyhella) [inline docs/troubleshooting/monitoring.md:137]:
```suggestion
Check the deployment and volume.
```

File: docs/troubleshooting/monitoring.md [hunk 6]
```diff
@@ -69,3 +69,107 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+# kubectl get volume -A
+NAMESPACE         NAME                                       STATE      ROBUSTNESS   SCHEDULED   SIZE          NODE     AGE
+longhorn-system   pvc-1b2fbbe9-14b1-4a65-941a-7d5645a89977   attached   degraded                 5368709120    harv31   43h
+longhorn-system   pvc-7c6dcb61-51a9-4a38-b4c5-acaa11788978   attached   degraded                 53687091200   harv31   43h
+longhorn-system   pvc-b2b2c07c-f7cd-4965-90e6-ac3319597bf7   detached   unknown                  2147483648             43h
+```
+
+### Expand Volume
+
+In Longhorn WebUI, the related volume becomes `Detached`. Click `Operation` colume menu, and select `Expand Volume`.
```
Comment (vickyhella) [inline docs/troubleshooting/monitoring.md:156]:
```suggestion
In the Longhorn WebUI, the related volume becomes `Detached`. Click the icon in the `Operation` column and select `Expand Volume`.
```

**Review by vickyhella — COMMENTED** [review]
Just some nits. BTW, I only added comments for the v1.1 doc as the v1.1 and v1.0 changes are the same. When applying the suggestions, please kindly do the same for the v1.0 doc.

File: docs/troubleshooting/monitoring.md [hunk 7]
```diff
@@ -69,3 +69,121 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+
+![](/img/v1.1/troubleshooting/4-select-volume-to-expand.png)
+
+Input a new size, and `Longhorn` will expand the volume to this size.
+
+![](/img/v1.1/troubleshooting/5-expand-volue-to-new-size.png)
+
+### Scale Up a Deployment
+
+To scale up a deployment to the original replicas:
```
Comment (vickyhella) [inline docs/troubleshooting/monitoring.md:166]: `to the original replicas` needs to be more detailed as mentioned before

File: versioned_docs/version-v1.0/troubleshooting/monitoring.md [hunk 8]
```diff
@@ -69,3 +69,121 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+
+![](/img/v1.0/troubleshooting/4-select-volume-to-expand.png)
+
+Input a new size, and `Longhorn` will expand the volume to this size.
+
+![](/img/v1.0/troubleshooting/5-expand-volue-to-new-size.png)
+
+### Scale Up a Deployment
+
+To scale up a deployment to the original replicas:
```
Comment (vickyhella) [inline versioned_docs/version-v1.0/troubleshooting/monitoring.md:166]: ditto

**Review by vickyhella — COMMENTED** [review]
Some missing suggestions.

File: docs/troubleshooting/monitoring.md [hunk 9]
```diff
@@ -69,3 +69,121 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
 
 ```
 
+## Expand PV/Volume Size
+
+`Harvester` integrates `Longhorn` as the default storage provider.
+
+Harvester `Monitoring` uses `Persistant Volume (PV)` to store running data. When a cluster has been running for a certain time, the `Persistant Volume` may need to expand its size.
+
```
Comment (vickyhella) [inline docs/troubleshooting/monitoring.md:77]:
```suggestion
Harvester `Monitoring` uses `Persistent Volume (PV)` to store running data. When a cluster has been running for a certain time, the `Persistent Volume` may need to expand its size.

```

Comment (vickyhella) [reply docs/troubleshooting/monitoring.md:77]: I found that some previous suggesions haven't been applied but marked as resolved. Please double-check~

File: versioned_docs/version-v1.0/troubleshooting/monitoring.md [hunk 10]
```diff
@@ -69,3 +69,121 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
 
 ```
 
+## Expand PV/Volume Size
+
+`Harvester` integrates `Longhorn` as the default storage provider.
+
+Harvester `Monitoring` uses `Persistant Volume (PV)` to store running data. When a cluster has been running for a certain time, the `Persistant Volume` may need to expand its size.
```
Comment (vickyhella) [inline versioned_docs/version-v1.0/troubleshooting/monitoring.md:76]:
```suggestion
Harvester `Monitoring` uses `Persistent Volume (PV)` to store running data. When a cluster has been running for a certain time, the `Persistent Volume` may need to expand its size.
```

File: docs/troubleshooting/monitoring.md [hunk 11]
```diff
@@ -69,3 +69,121 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+Based on the `Longhorn` `Volume` expansion guide, `Harvester` illustrates how to [expand the volume size](https://longhorn.io/docs/1.3.2/volumes-and-nodes/expansion/).
+
+### View Volume
+
+#### From Embedded Longhorn WebUI
+
+Access the embedded Longhorn WebUI according to [this document](https://docs.harvesterhci.io/v1.0/troubleshooting/harvester/#access-embedded-longhorn).
```
Comment (bk201) [inline docs/troubleshooting/monitoring.md:84]: We could use a relative path here.

File: docs/troubleshooting/monitoring.md [hunk 12]
```diff
@@ -69,3 +69,121 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+rancher-monitoring-grafana              1/1     1            1           43h  // target deployment
+rancher-monitoring-kube-state-metrics   1/1     1            1           43h
+rancher-monitoring-operator             1/1     1            1           43h
+rancher-monitoring-prometheus-adapter   1/1     1            1           43h
+```
+
+Scale down the deployment `rancher-monitoring-grafana` to 0.
+
+```
+# kubectl scale --replicas=0 deployment/rancher-monitoring-grafana  -n cattle-monitoring-system
```
Comment (bk201) [inline docs/troubleshooting/monitoring.md:134]:
```suggestion
# kubectl scale --replicas=0 deployment/rancher-monitoring-grafana -n cattle-monitoring-system
```
(nit).

File: docs/troubleshooting/monitoring.md [hunk 13]
```diff
@@ -69,3 +69,121 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+Input a new size, and `Longhorn` will expand the volume to this size.
+
+![](/img/v1.1/troubleshooting/5-expand-volue-to-new-size.png)
+
+### Scale Up a Deployment
+
+To scale up a deployment to the original replicas:
+
+```
+# kubectl scale --replicas=1 deployment/rancher-monitoring-grafana  -n cattle-monitoring-system
```
Comment (bk201) [inline docs/troubleshooting/monitoring.md:169]:
```suggestion
# kubectl scale --replicas=1 deployment/rancher-monitoring-grafana -n cattle-monitoring-system
```
ditto.

**Review by bk201 — COMMENTED** [review]
The procedure works for me, I can expand a Grafana PVC.

File: docs/troubleshooting/monitoring.md [hunk 14]
```diff
@@ -69,3 +69,121 @@ rancher-monitoring-grafana-d9c56d79b-cp86w               3/3     Running   0
...
+# kubectl get volume -A
+NAMESPACE         NAME                                       STATE      ROBUSTNESS   SCHEDULED   SIZE          NODE     AGE
+longhorn-system   pvc-1b2fbbe9-14b1-4a65-941a-7d5645a89977   attached   degraded                 5368709120    harv31   43h
+longhorn-system   pvc-7c6dcb61-51a9-4a38-b4c5-acaa11788978   attached   degraded                 53687091200   harv31   43h
+longhorn-system   pvc-b2b2c07c-f7cd-4965-90e6-ac3319597bf7   attached   degraded                 2147483648    harv31   43h
+```
+
+### Scale Down a Deployment
+
+To detach the `Volume`, you need to scale down the `deployment` which uses the `Volume`.
```
[the remark this answers is not in the document]
Comment (w13915984028) [reply docs/troubleshooting/monitoring.md]: Question: what's the difference between `that` and `which` here?

File: docs/troubleshooting/monitoring.md (hunk 7 above)
Comment (w13915984028) [reply docs/troubleshooting/monitoring.md]: tried to add more details, please check again.

BTW: A common understanding is that the `trouble-shooting` is targeting those technical users, when they have certain background of `kubernetes`, they will not be confused by `replica` concept.

File: docs/troubleshooting/monitoring.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (vickyhella) [reply docs/troubleshooting/monitoring.md]: I think `that` is more grammatically correct ^^

File: docs/troubleshooting/monitoring.md (hunk 7 above)
Comment (vickyhella) [reply docs/troubleshooting/monitoring.md]: Can we say: To scale up a deployment to `1`:
I remember suggesting it in yesterday's comment, but I'm not sure if my understanding is correct

Comment (w13915984028) [reply docs/troubleshooting/monitoring.md]: The `scale up` is the reverse operation of `scale down`, it needs to keep the number of `replicas` as the original values. We use `the original replicas` to highlight this key point.

Please take a look of the updated description, thanks.

**Review by bk201 — APPROVED** [review]
LGTM!

**Review by vickyhella — APPROVED** [review]
LGTM and thanks!!

**Review by guangbochen — APPROVED** [review]

**Pull request merged**